shadows/framework/src/main/java/org/robolectric/shadows/ImageUtil.java

```diff
@@ -85,13 +126,40 @@ public static boolean writeToStream(
   private static String getFormatName(CompressFormat compressFormat) {
     switch (compressFormat) {
       case JPEG:
+        return FORMAT_NAME_JPEG;
       case WEBP:
       case WEBP_LOSSY:
       case WEBP_LOSSLESS:
-        return "jpg";
       case PNG:
-        return "png";
+        return FORMAT_NAME_PNG;
     }
     throw new UnsupportedOperationException("Cannot convert format: " + compressFormat);
   }
+
+  private static boolean needAlphaChannel(CompressFormat compressFormat) {
+    return !FORMAT_NAME_JPEG.equals(getFormatName(compressFormat));
+  }
+
+  private static int getBufferedImageType(Bitmap.Config config, boolean needAlphaChannel) {
+    if (config == null) {
+      return needAlphaChannel ? TYPE_INT_ARGB : TYPE_INT_RGB;
+    }
+    switch (config) {
+      case RGB_565:
+        return BufferedImage.TYPE_USHORT_565_RGB;
+      case RGBA_F16:
+        return needAlphaChannel ? TYPE_INT_ARGB_PRE : TYPE_INT_RGB;
+      case ALPHA_8:
+      case ARGB_4444:
+      case ARGB_8888:
+      case HARDWARE:
+      default:
+        return needAlphaChannel ? TYPE_INT_ARGB : TYPE_INT_RGB;
+    }
+  }
+
+  static class RobolectricBufferedImage {
```

Comment:
This class seems overkill -- can't the mime type just be read from BufferedImage?

Reply:
The `BufferedImage` doesn't have mime type.

Reply:
Looked at the BufferedImage and realized this is not possible, so this is fine with me.

Reply:
There is a StackOverflow question about it: https://stackoverflow.com/questions/21057191/can-i-tell-what-the-file-type-of-a-bufferedimage-originally-was.